backend/src/backend/tests.py

```diff
@@ -50,10 +50,36 @@ def test_chat_endpoint_malformed_request():
     assert "msg" in response.text
 
 
-# Test for empty `message` field
 def test_chat_endpoint_empty_message():
     response = client.post("/api/chat", json={"message": ""})
     assert response.status_code == 400
+    # TODO: Add more specific error message
     assert "type" in response.text
     assert "loc" in response.text
-    assert "msg" in response.text
\ No newline at end of file
+    assert "msg" in response.text
+
+
+def test_chat_endpoint_success(mocker):
+    """
+    Testing features with external API calls can be done in several ways:
+    1. Setting up a test environment for the external API.
+    2. Mocking the external API call to return a known response.
+
+    Mocking is the simplest and most efficient method. It isolates the test
+    from the external API, making it faster and more reliable.
+
+    This test demonstrates how to mock the OpenAI API call within the chat
+    endpoint to return a known response.
+    """
+    mocker.patch("openai.api_key", return_value="test")
+    openai_mock_response = mocker.MagicMock()
+    openai_mock_response.choices = [mocker.MagicMock()]
+    openai_mock_response.choices[0].message.content = "What's up?"
+    mocker.patch(
+        "openai.chat.completions.create",
+        return_value=openai_mock_response,
+    )
+
+    response = client.post("/api/chat", json={"message": "Hello, World!"})
+    assert response.status_code == 200
+    assert response.json() == {"response": "What's up?"}
```

Comment:
This is enough to test the successful scenario. 
But we should consider the cases where the Open AI endpoint returns an error code. Also, make sure to handle the scenario where an Open AI call hits the limit (typically Chat GPT has a limit). The existing code is not handling the case that the user is using the feature too much. 
Since this API costs, we should be careful with it. Probably tracking user & # request per hour to start?

Also, the live code needs request validation.

Reply:
I agree. We should handle errors such as the rate limit being exceeded.
Since this pull request only aims to demonstrate how the Python mock works and add a test for the simplest scenario, I'm not considering tackling it in this pull request. Can you please create an issue and assign it to, maybe @LucileTech? Thanks!

Also, what do mean by "request validation"? LLM guardrails?

Reply:
For request validation, we need to make sure that
1. message in request is not empty.
2. message is not too long likely hitting rate limit easily.

Reply:
Thanks for the suggestion 👍 

We already have a validation for the first one. As you can see in this test, an empty request results in 400 Bad Request.
 https://github.com/mie-h/sprout/pull/57/files#diff-babc46e97ac405001cee90dbe6f0cd12141adb70a20ad4f397448a0e0982980aR53

For the second point, let me create an issue, though we are still not sure how we interact with the API and how many tokens we need.

Reply:
created https://github.com/mie-h/sprout/issues/60

Reply:
Sorry, I didn't explain enough. My point was that when validating, we also need to return more detailed response. This would help debugging process.